stat.c

```diff
@@ -552,7 +552,7 @@ static void show_ddir_status(struct group_run_stats *rs, struct thread_stat *ts,
 			bw_str = "kB";
 
 		if (rs->agg[ddir]) {
-			p_of_agg = mean * 100 / (double) (rs->agg[ddir] / 1024);
+			p_of_agg = mean * 100 / (double) (rs->agg[ddir] / 1024.0);
```

Comment:
This one is tricky. Before we did integer arithmetic to convert rs->agg[ddir] into kilobytes which mean it could just come out to 0. Arguably the if above was trying to catch that but it dates from a time before this conversion was happening. Maybe it would be better to check if rs->agg[ddir] was less than 1024 before going into the body. There's also similar code around line 1380 so maybe we should turn this into a function?

Reply:
Done. Create a function